primitives/staking/src/offence.rs

```diff
@@ -142,19 +142,28 @@ pub trait OnOffenceHandler<Reporter, Offender> {
 	/// Zero is a valid value for a fraction.
 	///
 	/// The `session` parameter is the session index of the offence.
+	///
+	/// The receiver might decide to not accept this offence. In this case, the call site is
+	/// responsible for queuing the report and re-submitting again.
 	fn on_offence(
 		offenders: &[OffenceDetails<Reporter, Offender>],
 		slash_fraction: &[Perbill],
 		session: SessionIndex,
-	);
+	) -> Result<(), ()>;
+
+	/// Can an offence be reported now or not? This should always be false if a direct call to
+	/// [`on_offence`] was to return `Err`.
```

Comment:
We can't really guarantee that, so I think the comment here should rather say that this should only be considered an optimization.
Also it seem that passing the arguments (or at least `OffenceDetails`) would be useful for some cases, I'd consider including that too.

Reply:
Yes exactly it is just an optimization. Arguments will be unused for now and since I see this whole PR as potentially temporary, then I won't add it for now.

Reply:
Agree with Tomek that there may be false negatives (in this case `true` return values).